src/storage-client/src/controller.rs

```diff
@@ -1010,6 +1022,26 @@ where
                             self.truncate_managed_collection(id).await;
                             self.initialize_shard_mapping().await;
                         }
+                        IntrospectionType::StorageHostMetrics => {
+                            // The metrics scraper assumes that the collection
+                            // sums up to empty when starting up. When needed,
+                            // we can change the scraper to reconcile its
+                            // internal state with the current state of the
+                            // output collection.
+                            self.truncate_managed_collection(id).await;
+
+                            let metrics_fetcher = self.hosts.metrics_fetcher();
+                            let scraper_token = metrics_scraper::spawn_metrics_scraper(
+                                id.clone(),
+                                metrics_fetcher,
+                                // This does a shallow copy.
+                                self.state.collection_manager.clone(),
+                            );
+
+                            // Make sure this is dropped when the controller is
+                            // dropped, so that the internal task will stop.
```

Comment:
when is the controller dropped, but we arent completely shutting down?

Reply:
I think it never is, but this makes sure that we're nice and tidy about ourselves if there ever is a future where controllers can be dropped or we might want to drop the scraper